tests/unit_tests/dagmc/test_model.py

```diff
@@ -254,3 +258,111 @@ def test_dagmc_xml(model):
+    settings = openmc.Settings()
+    settings.batches = 100
+    settings.particles = 1000
+    model = openmc.Model()
+    model.materials = openmc.Materials(mats.values())
+    model.geometry = openmc.Geometry(root=daguniv)
+    model.settings = settings
+    model.init_lib()
+    model.sync_dagmc_universes()
+    for univ in model.geometry.get_all_universes().values():
+        if isinstance(univ, openmc.DAGMCUniverse):
+            dag_univ = univ
+            break
+
+    my_tally = openmc.Tally()
+    cell_filter = openmc.CellFilter([x.id for x in dag_univ.get_all_cells().values()])
+
+    my_tally.scores = ['nu-fission']
+    my_tally.filters = [cell_filter]
+    model.tallies.append(my_tally)
+    src = openmc.IndependentSource(space=openmc.stats.Point())
+    model.settings.source = src
+
+    model.export_to_model_xml('sub_dir_not_a_vacuum/model.xml',)
+    openmc.run(cwd='sub_dir_not_a_vacuum')
+    # Read the statepoint file.
+    sp = openmc.StatePoint("sub_dir_not_a_vacuum/statepoint.100.h5")
+
+    # Extract the tally data as a Pandas DataFrame.
+    tally_dfs = [t.get_pandas_dataframe() for t in sp.tallies.values()]
+    df = pd.concat(tally_dfs, ignore_index=True)
+    # not detected as vacuum so tally should be non-zero
+    assert df['mean'].loc[0] > 0
+
+    shutil.copyfile(Path(request.fspath).parent / "dagmc.h5m", tmpdir/"dagmc_a_vacuum.h5m")
+    hf = h5py.File(tmpdir/ "dagmc_a_vacuum.h5m", 'r+')
+    new_assignment = 'mat:vacuum'.encode('utf-8')
+    hf['/tstt/tags/NAME']['values'][1] = new_assignment
+    hf.close()
+
+    p = tmpdir/"dagmc_a_vacuum.h5m"
+    daguniv = openmc.DAGMCUniverse(p, auto_geom_ids=True).bounded_universe()
+    settings = openmc.Settings()
+    settings.batches = 100
+    settings.inactive = 10
+    settings.particles = 1000
+    model = openmc.Model()
+    model.materials = openmc.Materials(mats.values())
+    model.geometry = openmc.Geometry(root=daguniv)
+    model.settings = settings
+    model.init_lib()
+    model.sync_dagmc_universes()
+    for univ in model.geometry.get_all_universes().values():
+        if isinstance(univ, openmc.DAGMCUniverse):
+            dag_univ = univ
+            break
+
+    my_tally = openmc.Tally()
+    cell_filter = openmc.CellFilter([x.id for x in dag_univ.get_all_cells().values()])
+
+    my_tally.scores = ['nu-fission']
+    my_tally.filters = [cell_filter]
+    model.tallies.append(my_tally)
+    src = openmc.IndependentSource(space=openmc.stats.Point())
+    model.settings.source = src
+
+    model.export_to_model_xml('sub_dir_vacuum/model.xml',)
+    openmc.run(cwd='sub_dir_vacuum')
+    # Read the statepoint file.
+    sp = openmc.StatePoint("sub_dir_vacuum/statepoint.100.h5")
+
+    # Extract the tally data as a Pandas DataFrame.
+    tally_dfs = [t.get_pandas_dataframe() for t in sp.tallies.values()]
+    df = pd.concat(tally_dfs, ignore_index=True)
+    # detected as vacuum so tally should be zero
+    assert df['mean'].loc[0] == 0
```

Comment:
I don't think we need pandas here

```suggestion
    tally_dfs = [t.get_pandas_dataframe() for t in sp.tallies.values()]
    df = pd.concat(tally_dfs, ignore_index=True)
    # detected as vacuum so tally should be zero
    assert df['mean'].loc[0] == 0
```